### PR TITLE
Update DXF export section

### LINKED ISSUE #8998: [dxf] Import/export 'DXF Export' dialog settings from/to XML (Request in QGIS)
### Request for documentation
From pull request qgis/QGIS#56982
Author: @gacarrillor
QGIS version: 3.38

**[dxf] Import/export 'DXF Export' dialog settings from/to XML**

### PR Description:
This PR enables users to save and restore GUI settings for the DXF Export dialog, making it possible to export a number of configurations and reuse them or share them with colleagues.

Settings are exported to an XML file (see [sample file content](https://github.com/qgis/QGIS/files/14771343/dxf_settings.xml.txt)).

https://github.com/gacarrillor/QGIS/assets/652785/0ea63c1b-47cf-4c64-a475-c6b3c549d8b8

Buttons to save/load to/from DXF Export settings file:

image

Confirmation message before overriding dialog settings with those present in the XML file:

image

Non-blocking confirmation of settings applied:

image

-----
Funded by the [QGIS user group Switzerland](https://qgis.ch/).

### Commits tagged with [need-docs] or [FEATURE]

### LINKED ISSUE #9028: [dxf] Allow users to override the layer name to be exported to DXF (Request in QGIS)
### Request for documentation
From pull request qgis/QGIS#57016
Author: @gacarrillor
QGIS version: 3.38

**[dxf] Allow users to override the layer name to be exported to DXF**

### PR Description:
This PR makes it possible to override the output name of individual layers exported in the DXF Export dialog.

The corresponding processing algorithm's GUI and parameter were updated accordingly.

Tests were added for:
   + Output layer name precedence for the exported DXF: 1) attribute, 2) layer title, 3) overridden layer name, 4) layer name.
   + Processing `qgsprocessingparameterdxflayers` parameter and DXF layer's widget wrapper.

Screencast App dialog

https://github.com/gacarrillor/QGIS/assets/652785/fbb3443e-24fe-45b6-b735-43d3b4abdea5

Screencast Processing dialog

https://github.com/gacarrillor/QGIS/assets/652785/d94e3ce4-a289-4888-88c5-4557259dc3e2

UX hints for the users:

DXFExport_algs_help
DXFExport_tooltip
DXFExport_alg_param_tooltip

-----------------

Funded by the [QGIS user group Switzerland](https://qgis.ch/).

### Commits tagged with [need-docs] or [FEATURE]

### LINKED ISSUE #9045: [processing] Harmonize DXF Export app dialog and corresponding algorithm (Request in QGIS)
### Request for documentation
From pull request qgis/QGIS#57200
Author: @gacarrillor
QGIS version: 3.38

**[processing] Harmonize DXF Export app dialog and corresponding algorithm**

### PR Description:
Follow-up: https://github.com/qgis/QGIS/pull/56147

 + Remember settings for data defined blocks in DXF Export app dialog 
 + Expose DXF Export's data defined blocks functionality to processing

image

Fix https://github.com/qgis/QGIS/issues/56853

-----------------

Funded by the [QGIS user group Switzerland](https://qgis.ch/).

### Commits tagged with [need-docs] or [FEATURE]

### PR DESCRIPTION
**Pull request by gacarrillor**
+ [x] Add paragraph for export/import DXF export settings to/from XML
 + [x] Rename output layer name.
 + [x] Add details for `Input layer` parameter in `Export layers to DXF` alg.

Fix #8998
Fix #9028 
Fix #9045

### CODE REVIEW

**Comment by gacarrillor** [comment]
@DelazJ

I've added a little note on the processing alg to mention new options in "Input layers" parameter.

I was wondering if that's enough to also consider #9045 fixed or if we should add a screenshot like the one you can find in #9045's description.

**Comment by DelazJ** [comment]
I'm not sure we need a screenshot but I'd rather see something in the vein of what was done for align rasters "[layer inputs](https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/qgis/rastertools.html#align-rasters)", i.e. how you get to the options and what they really are (name, type, default, ... information that helps to call them in a code, as any other parameter)

File: docs/user_manual/managing_data_source/create_layers.rst [hunk 1]
```diff
@@ -541,6 +541,26 @@ Optionally, you can also choose to:
...
+:guilabel:`Settings` combo box has two options: :guilabel:`Load Settings from
+File...` and :guilabel:`Save Settings to File...`.
+
+.. tip:: Override the layer name to be exported to DXF
+
+   Output DXF layer names may be overridden without altering the original
+   project layers. For this, click on a layer name on the dialog's layers panel
+   and set the output layer name.
+
+.. note::
```
Comment (DelazJ) [inline docs/user_manual/managing_data_source/create_layers.rst:555]:
```suggestion
```
what about merging these?

**Comment by gacarrillor** [comment]
> I'm not sure we need a screenshot but I'd rather see something in the vein of what was done for align rasters "[layer inputs](https://docs.qgis.org/testing/en/docs/user_manual/processing_algs/qgis/rastertools.html#align-rasters)", i.e. how you get to the options and what they really are (name, type, default, ... information that helps to call them in a code, as any other parameter)

Looks great, thanks for the pointer.

File: docs/user_manual/managing_data_source/create_layers.rst (hunk 1 above)
Comment (gacarrillor) [reply docs/user_manual/managing_data_source/create_layers.rst:555]: The precedence note is useful even if users don't use the overridden name (which we show in the tip). For instance, when using an attribute and enabling the `Use title` checkbox as well.
If we merge them, I'm afraid the precedence would be a bit more difficult to spot.

Nonetheless, if after reading the paragraph above you still feel it's better to merge them, we'll do it. You have the overall picture here.

Comment (DelazJ) [reply docs/user_manual/managing_data_source/create_layers.rst:555]: Yes I agree, the precedence detail is very important, even more than the override tip imho.
How about keeping the note at the end but move the renaming feature to line 523? We are somehow explaining the "layer attribute" column there and will need to explain the block things in a near future, so let's do it for the first column also. I mean something like:
> For each layer, you can:
> * the tip about the name overriding
> * the split info using the layer attribute (and maybe make it clearer that the output layers take the field name)

Comment (gacarrillor) [reply docs/user_manual/managing_data_source/create_layers.rst:556]: Much better, thanks for the suggestion. Done in https://github.com/qgis/QGIS-Documentation/pull/9095/commits/b3695922189406200b3fe1eb96f20686c567ff06

**Pull request merged**